Skip building XS when PUREPERL_ONLY=1 set

zmughal opened the pull request:
This change skips the static/dynamic sections of the Makefile when using a
pure-Perl install so that no XS linking is done at all.

Compare the build logs below:

```shell
$ for br in master pureperl-skip-linking; do \
echo "On branch $br"; \
git checkout -q $br ; \
git clean -f; \
( perl Makefile.PL PUREPERL_ONLY=1 && make && make clean)  | grep -Fe $( perl -MConfig -e 'print $Config{cc}' ); \
echo; \
done
```

```
On branch master
Checking for cc... cc
cc -c   -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -O2   -DVERSION=\"1.102\" -DXS_VERSION=\"1.102\" -fPIC "-I$HOME/perl5/perlbrew/perls/perl-5.26.1/lib/5.26.1/x86_64-linux/CORE"   Util.c
cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong  Util.o  -o blib/arch/auto/Params/Util/Util.so  \

On branch pureperl-skip-linking
Checking for cc... cc
```

zmughal committed:
Skip linking XS code for pure Perl install

When `PUREPERL_ONLY=1` set, this prevents the Makefile from having rules
that compile the XS code. This is useful to have when building
dependencies for deploying to a pure Perl install such as when using
`App::FatPacker`. This removes the need to skip over files that end in
`perl -V:dlext` (e.g., `.so`, `.dll`, `.dylib`) when copying to a
deployment.

This should allow for using the `cpanm --pureperl` option so that a pure
Perl installation can be requested for all installed distributions.

Signed-off-by: Zakariyya Mughal <zaki.mughal@gmail.com>

zmughal commented:
I see there is not a CI configuration running any more (previously Travis CI). Would it help to use GitHub Actions to test this out?

rehsack commented:
I would love a PR which test whether this provides a reasonable and working change :)
Further, I'd prefer commit messages which explain the business case behind a commit, not the technical change. Most people reading commits can derive technical stuff on it's own and the remaining need the business reasons.

zmughal committed:
Add GitHub Actions workflow for testing various configurations

This is to test that the tests pass as expected under various versions
of Perl both using XS and pure Perl installs.

Signed-off-by: Zakariyya Mughal <zaki.mughal@gmail.com>

zmughal commented:
@rehsack, OK, sounds good!

You can review updated commits to see if this fits the requirements. A workflow that tests what I'm getting at is [here](https://github.com/zmughal/Params-Util/actions/runs/879680925).

zmughal commented:
Hi, if you have time, could you review this. It should be working on multiple platforms per the new CI workflow.

zmughal commented:
@rehsack, just reaching out again about this PR.

zmughal commented:
@rehsack, hello, do you have a moment to review this PR?

zmughal commented:
Hello @rehsack, I just ran the CI I made here again and it still passes. I also added a few more Perl versions to test against.

plicease commented:
can confirm that this one line patch makes it work.  What is currently on CPAN does not.

rehsack commented:
@plicease / @zmughal - beside merging that: I will not re-install perl and make a release.
I spent to much spare time with negative return, so I decided to avoid future invest of spare time.

Neither Perl nor CPAN have any relevant role on embedded platforms.

zmughal commented:
Thank you for merging, @rehsack. If there is anything that I can do to help, please let me know. Also, thank you for looking at the change, @plicease.

I have made the CI green again with #4.

plicease commented:
@rehsack How about giving @zmughal (CPAN id ZMUGHAL) and myself co-maint (or even owner) we can cut the release if you are no longer interested in maintaining this module.  My CPAN id is PLICEASE.

plicease commented:
If you do not have access to PAUSE or the time to do it yourself you can also email the PAUSE admins at modules@perl.org and ask them to do it for you, they will be only to happy to do so.

rehsack commented:
@plicease & @zmughal - we can probably talk about that. But I would expect that XS remains supported on host as well for cross-compile at the same level as the PP support is.

plicease commented:
> @plicease & @zmughal - we can probably talk about that. But I would expect that XS remains supported on host as well for cross-compile at the same level as the PP support is.

XS doesn't work for a fat packed script (`App::FatPacker`) which is the application that I am working on.  While I appreciate your having merged this PR, it doesn't really help me without a corresponding release since the system I am using gets modules from CPAN.

I'm not sure how this is relevant to the discussion of the ownership/co-maint status though, you yourself said that you are not interested in maintaining this module, why not let us take it off your hands?
